### ISSUE TITLE
tensor size mismatch problem

### ISSUE THREAD

**Issue by Adenialzz** [instantX-research/Regional-Prompting-FLUX#10]
Hi, thanks for your open-sourcing. Can you help to solve this problem ?

![image](https://github.com/user-attachments/assets/186782a1-962e-4ea1-8a54-a8ed1c4c25ca)

**Comment by antonioo-c** [comment]
Which diffuser version are you using? Could you provide the regional setups that you are using as well? Thank you.

**Comment by Adenialzz** [comment]
I reloaded the model and the code worked fine. So every time I want to change the mask area, I have to reload the model (I used jupyter notebook to avoid model reloading to save time), otherwise the above problem occurs.

**Comment by Adenialzz** [comment]
```python

import torch
from pipeline_flux_regional import RegionalFluxPipeline, RegionalFluxAttnProcessor2_0

flux_path = '/home/jeeves/JJ_Projects/klara_models/flux_dev/train-model'
pipeline = RegionalFluxPipeline.from_pretrained(flux_path, torch_dtype=torch.bfloat16).to("cuda")
attn_procs = {}
for name in pipeline.transformer.attn_processors.keys():
    if 'transformer_blocks' in name and name.endswith("attn.processor"):
        attn_procs[name] = RegionalFluxAttnProcessor2_0()
    else:
        attn_procs[name] = pipeline.transformer.attn_processors[name]
pipeline.transformer.set_attn_processor(attn_procs)

```

**Comment by Adenialzz** [comment]
I have to run the following code every time I change mask region

```python

import torch
from pipeline_flux_regional import RegionalFluxPipeline, RegionalFluxAttnProcessor2_0

flux_path = '/path/to/flux_dev'
pipeline = RegionalFluxPipeline.from_pretrained(flux_path, torch_dtype=torch.bfloat16).to("cuda")
attn_procs = {}
for name in pipeline.transformer.attn_processors.keys():
    if 'transformer_blocks' in name and name.endswith("attn.processor"):
        attn_procs[name] = RegionalFluxAttnProcessor2_0()
    else:
        attn_procs[name] = pipeline.transformer.attn_processors[name]
pipeline.transformer.set_attn_processor(attn_procs)

```

**Comment by antonioo-c** [comment]
@Adenialzz The mismatch problem is due to the regional mask cache in each attention processor (https://github.com/instantX-research/Regional-Prompting-FLUX/blob/5ce6e6ee1cf897268cbc208ed04d4717055b229c/pipeline_flux_regional.py#L266).
So in order to use regional prompting without reloading the whole model:

you can either add this code before applying each new regional control

```python
for processor in pipeline.transformer.attn_processors.values():
    if processor.regional_mask is not None:
        processor.regional_mask = None
````

or you can just comment the caching line and use passed-in regional mask.

```python
if base_ratio is not None and 'regional_attention_mask' in additional_kwargs:
        regional_mask = additional_kwargs['regional_attention_mask'].to(hidden_states.device)
else:
        regional_mask = None
```

**Comment by Adenialzz** [comment]
thanks, it makes sense. But I wonder why we need to cache the mask region? is it assumed that the user always use the same mask?

**Comment by antonioo-c** [comment]
It slightly speeds up the inference.
